### PR TITLE
feat: add oidc action test infra

### PR DESCRIPTION
**Pull request by bharathkkb**

### CODE REVIEW

File: actions-test-infra/oidc-auth-google-cloud/wif.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  source  = "terraform-google-modules/github-actions-runners/google//modules/gh-oidc"
+  version = "~> 1.1"
+
+  project_id  = var.gcp_project
+  pool_id     = "test-pool"
+  provider_id = "test-gh-provider"
+  sa_mapping = {
+    (google_service_account.oidc-auth-test-sa.account_id) = {
+      sa_name   = google_service_account.oidc-auth-test-sa.name
+      attribute = "attribute.repository/sethvargo/oidc-auth-google-cloud"
```
Comment (bharathkkb) [inline actions-test-infra/oidc-auth-google-cloud/wif.tf:37]: TBD

File: actions-test-infra/oidc-auth-google-cloud/backend.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  backend "gcs" {
+    bucket = "actions-infra-tfstate-b4fa"
+    prefix = "state/oidc-auth-google-cloud"
```
Comment (sethvargo) [inline actions-test-infra/oidc-auth-google-cloud/backend.tf:20]: Should we rename it as "auth" on the move in?

File: actions-test-infra/oidc-auth-google-cloud/wif.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+module "project-services" {
+  source  = "terraform-google-modules/project-factory/google//modules/project_services"
+  version = "~> 9.0"
+
+  project_id = var.gcp_project
+  activate_apis = [
+    "sts.googleapis.com",
+    "iamcredentials.googleapis.com",
```
Comment (sethvargo) [inline actions-test-infra/oidc-auth-google-cloud/wif.tf:24]:
```suggestion
    "iamcredentials.googleapis.com",
    "secretmanager.googleapis.com",
```

File: actions-test-infra/oidc-auth-google-cloud/wif.tf (hunk 1 above)
Comment (bharathkkb) [reply actions-test-infra/oidc-auth-google-cloud/wif.tf]: switched to google-github-actions/auth

File: actions-test-infra/oidc-auth-google-cloud/backend.tf (hunk 2 above)
Comment (bharathkkb) [reply actions-test-infra/oidc-auth-google-cloud/backend.tf]: yup switched to `auth`

**Review by sethvargo — APPROVED** [review]

**Review by averikitsch — APPROVED** [review]

**Pull request merged**